api/test/models/checkout-items/checkout-items-data-mapper-impl.test.ts

```diff
@@ -0,0 +1,136 @@
+// GLOBAL REQUIREMENTS
+import { Substitute } from '@fluffy-spoon/substitute';
+import { expect } from 'chai';
+import 'mocha';
+import { CheckoutItems, CheckoutItemsDataMapperImpl } from '../../../lib/models/checkout-items';
+import { RBAC } from '../../../lib/services/auth/RBAC/rbac';
+import { IAcl } from '../../../lib/services/auth/RBAC/rbac-types';
+import { IDataMapper } from '../../../lib/services/database';
+import { MysqlUow } from '../../../lib/services/database/svc/mysql-uow.service';
+import { Logger } from '../../../lib/services/logging/logging';
+import { IActiveHackathonDataMapper } from '../../../lib/models/hackathon/active-hackathon';
+import { ActiveHackathon } from '../../../lib/models/hackathon/active-hackathon/active-hackathon';
+import { of } from 'rxjs';
+
+function mockedQuery<T>(query, params) {
+  // @ts-ignore
+  return Promise.resolve({ query, params });
+}
+
+let checkoutItemsDataMapper: IDataMapper<CheckoutItems>;
+let activeHackathonDataMapper;
+let mysqlUow;
+const acl: IAcl = new RBAC();
+
+describe('TEST: CheckoutItems Data Mapper', () => {
+  beforeEach(() => {
+    // Configure Mock MysqlUow
+    mysqlUow = Substitute.for<MysqlUow>();
+    // Configure Active Hackathon Data Mapper
+    activeHackathonDataMapper = Substitute.for<IActiveHackathonDataMapper>();
+    activeHackathonDataMapper.activeHackathon.returns(of(new ActiveHackathon({
+      basePin: 0,
+      endTime: null,
+      name: 'test hackathon',
+      uid: 'test uid',
+    })));
+    activeHackathonDataMapper.tableName.returns('HACKATHON');
+    // Configure CheckoutItems Data Mapper
+    checkoutItemsDataMapper = new CheckoutItemsDataMapperImpl(
+      acl,
+      mysqlUow,
+      activeHackathonDataMapper,
+      new Logger(),
+    );
+    // Configure mocked methods for mysql
+    mysqlUow.query().mimicks(mockedQuery);
+  });
+
+  describe('TEST: CheckoutItems get available', () => {
```

Comment:
I think this test should be the standard test for getting a specific item. The getAvailable and getAllAvailable should be different tests.